src/Compilers/CSharp/Portable/BoundTree/Formatting.cs

```diff
@@ -149,7 +149,15 @@ internal partial class BoundStackAllocArrayCreation
     {
         public override object Display
         {
-            get { return string.Format(MessageID.IDS_StackAllocExpression.Localize().ToString(), ElementType); }
+            get { return string.Format(MessageID.IDS_StackAllocExpression.Localize().ToString(), ElementType, Count.Syntax); }
+        }
+    }
+
+    internal partial class BoundConvertedStackAllocExpression
```

Comment:
Since a converted stack alloc expression has a type, it should not have any special `Display` implementation. Please remove.